src/Raven.Server/Documents/Replication/ReplicationLoader.cs

```diff
@@ -646,7 +646,20 @@ private TcpConnectionInfo GetConnectionInfo(ReplicationNode node, bool external)
                 }
                 if (node is InternalReplication internalNode)
                 {
-                    return ReplicationUtils.GetTcpInfo(internalNode.Url, internalNode.NodeTag, "Replication", _server.Server.Certificate.Certificate);
+                    var topology = ReplicationUtils.GetTopology(_server.Server.WebUrl, internalNode.Database, _server.Server.Certificate.Certificate);
+                    try
+                    {
+                        return ReplicationUtils.GetTcpInfo(internalNode.Url, internalNode.NodeTag, "Replication", _server.Server.Certificate.Certificate);
```

Comment:
Not the right place for this. This isn't something specific for replication.
It is generic for any type of error where we can access the topology but not the db.